packages/graphql-server/src/ee/utils/jobArtifactCleaner.ts

```diff
@@ -0,0 +1,118 @@
+      }
+
+      logger.info({type: 'JOB_ARTIFACT_CLEANUP_COMPLETED'});
+    } catch (e) {
+      logger.error({type: 'JOB_ARTIFACT_CLEANUP_FAILED', error: e});
+    }
+  }
+
+  // Start the timer to clean up the artifacts periodically
+  public start = async () => {
+    setInterval(this.cleanUp, CLEANUP_INTERVAL_SECONDS * 1000);
+  }
+
+  private listGroup = async (): Promise<string[]> => {
+    const prefix = `groups/`;
+    const groups = [];
+    return new Promise<string[]> ((resolve, reject) => {
+      const stream = this.minioClient.listObjects(this.bucket, prefix, false);
+      stream.on('data', (obj: BucketItem) => {
+        if (obj.prefix) {
+          const group = obj.prefix.split('/')[1];
+          groups.push(group);
+        }
+      });
+      stream.on('error', (error: Error) => {
+        reject(error);
+      });
+      stream.on('end', () => {
+        resolve(groups);
+      });
+    });
+  }
+
+  private cleanArtifactsByGroup = async group => {
+    const prefix = `groups/${group}/jobArtifacts/`;
+    const promises = [];
+    await new Promise<string[]> ((resolve, reject) => {
+      const stream = this.minioClient.listObjects(this.bucket, prefix, false);
+      stream.on('data', async (obj: BucketItem) => {
+        if (obj.prefix) {
+          const job = obj.prefix.split('/')[3];
+          promises.push(this.cleanArtifactsByJob(obj));
+        }
+      });
+      stream.on('error', (error: Error) => {
+        reject(error);
+      });
+      stream.on('end', () => {
+        resolve();
+      });
+    });
+
+    return Promise.all(promises);
+  }
+
+  private cleanArtifactsByJob = async jobObj => {
+    const job = jobObj.prefix.split('/')[3];
+    const expiredAt = await this.getExpiredAt(jobObj.prefix);
+    if (expiredAt && expiredAt < Date.now() / 1000) {
+      const objs = [];
+      logger.info({type: 'JOB_ARTIFACT_CLEAN_JOB_START', job});
+      try {
+        await new Promise<string[]> ((resolve, reject) => {
+          const stream = this.minioClient.listObjects(this.bucket, jobObj.prefix, true);
+          stream.on('data', async (obj: BucketItem) => {
+            objs.push(obj.name);
+          });
+          stream.on('error', (error: Error) => {
+            reject(error);
+          });
+          stream.on('end', () => {
+            resolve();
+          });
+        });
+
+        await this.minioClient.removeObjects(this.bucket, objs);
+        logger.info({type: 'JOB_ARTIFACT_CLEAN_JOB_COMPLETED', job});
+      } catch (e) {
+        logger.error({type: 'JOB_ARTIFACT_CLEAN_JOB_FAILED', job, error: e});
+      }
+    }
+  }
+
+  private getExpiredAt = async (jobPrefix): Promise<number> => {
+    const expiredAtFile = `${jobPrefix}.metadata/expiredAt`;
+    try {
+      const stream: Stream = await this.minioClient.getObject(this.bucket, expiredAtFile);
+      const result = await getStream(stream);
+      return Number(result);
+    } catch (e) {
+      // file not found
```

Comment:
Should we print a line of log to record this group has no expiredAtFile?

Reply:
In current design,
1. There is start/complete for each scheduled deletion.
1. If there is deletion required, it will print start/complete for one job deletion

So if I think it is enough for debugging. or you think it would be better to know how many groups we have traversed 
in?

Reply:
Done

```
{"time":"2020-10-13T01:23:37.995Z","level":"INFO","type":"JOB_ARTIFACT_CLEANUP_START"}
packages/graphql-server/src/logger.ts:34
{"time":"2020-10-13T01:23:38.625Z","level":"INFO","type":"JOB_ARTIFACT_CLEAN_GROUP_COMPLETED","group":"phusers","jobDeleted":0}
packages/graphql-server/src/logger.ts:34
{"time":"2020-10-13T01:23:39.048Z","level":"INFO","type":"JOB_ARTIFACT_CLEAN_GROUP_COMPLETED","group":"test","jobDeleted":0}
packages/graphql-server/src/logger.ts:34
{"time":"2020-10-13T01:23:39.051Z","level":"INFO","type":"JOB_ARTIFACT_CLEANUP_COMPLETED"}
```